### ISSUE TITLE
Unable to change parent categories for translated children. Previous modification reverts back to previous parent.

### ISSUE THREAD

**Issue by johnearnshaw** [polylang/polylang#327]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [x] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [x] I can reproduce this bug consistently

## Steps to reproduce the issue

1. Create 2 languages `(language-1, language-2)`
2. Post category: Create 2 parent categories for `language-1` and 2 parent categories for `language-2`
    `(language-1/parent-1) (language-1/parent-2) (language-2/parent-1) (language-2/parent-2)`
3. Post category: Create 1 child category for `language-1` and 1 child category for `language-2`
    `(language-1/parent-1/child-1) (language-2/parent-1/child-1)`. These categories must be polylang
    linked. (Eg. `language-2/parent-1/child-1` is the translated category for `language-1/parent-1/child-1`)
4. Post: Create 1 `new post` for `language-1` and assign the category to `child-1` and 1 `new post` for
    `language-2` and assign the category to `child-1`.
5. Edit category: Edit `language-1/child-1` and change parent category form `parent-1` to `parent-2`
6. Edit category: Edit `language-2/child-1` and change parent category from `parent-1` to `parent-2`
7. Edit category: Edit `language-1/child-1` and you will see that the parent category has returned back to
    `parent-1`
5. Edit category: Edit `language-1/child-1` and change parent category form `parent-1` to `parent-2`
9. Edit category: Edit `language-2/child-1` and you will see that the parent category has returned back to
    `parent-1`

## Expected behavior and actual behavior

When I follow those steps, I see...

When changing the parent category of a child category the parent category updates correctly. When modifying the parent category of the translated child category, the previous changed parent category returns to its old parent.

This also happens when you perform the action the opposite way around. Eg. Modify the parent category of the translated category first, before modifying the parent category for the primary category. Now the parent category for the translated child reverts back to the previous parent. This means that I'm unable to change parent categories for translated child categories. The previous modification reverts back to the previous parent.

I was expecting...

I should be able to change the parent category for all translated (linked) child categories.

**Comment by Chrystll** [comment]
Thank you for you detailed bug report. I was able to reproduce it. @Chouby  I found that this bug has been introduced in the 2.3-dev version probably by "Refactor term parent synchronization"[c8fddd5a148174b4e0ada6ee183ca09e40da40da](https://github.com/polylang/polylang/commit/c8fddd5a148174b4e0ada6ee183ca09e40da40da)
The previous "Refactor copy/sync of taxonomy terms" [4b9dd7967001aa07f89fb3aa4c19b907dee7daf3] (https://github.com/polylang/polylang/commit/4b9dd7967001aa07f89fb3aa4c19b907dee7daf3) commit triggers a fatal error.
